### PR TITLE
Perfect perfect-numbers

### PR DESCRIPTION
**Pull request by cmcaine**

### CODE REVIEW

**Comment by cmcaine** [comment]
The tests are failing because I'm relying on a feature that's in Julia 1.4 so that I can sum over an empty collection.

File: exercises/perfect-numbers/example.jl [hunk 1]
```diff
@@ -1,6 +1,10 @@
 function aliquot_sum(n)
     if !isinteger(n) || n < 0
         throw(DomainError(n,"n must be a non-negative integer"))
+    # In Julia <1.4, you need this special case because `sum` doesn't like
```
Comment (SaschaMann) [inline exercises/perfect-numbers/example.jl:4]: A workaround would be `reduce(+, i for i in 1:div(n, 2) if n % i == 0; init=0)` but it's slightly slower.

Comment (cmcaine) [reply exercises/perfect-numbers/example.jl:4]: I want to keep the workaround separate to emphasise that there is a nice option for modern Julias.

**Comment by cmcaine** [comment]
Might change this to be defining predicates instead.

**Pull request merged**